metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1769,19 +1772,22 @@ def _build_aggregated_measure_from_measure_source_node(
             )
 
         # If querying an offset metric, join to time spine before aggregation.
-        if before_aggregation_time_spine_join_description and base_queried_agg_time_dimension_specs:
+        if before_aggregation_time_spine_join_description and queried_agg_time_dimension_specs:
             unaggregated_measure_node = self._build_time_spine_join_node_for_offset(
                 join_description=before_aggregation_time_spine_join_description,
                 measure_properties=measure_properties,
-                base_queried_agg_time_dimension_specs=base_queried_agg_time_dimension_specs,
+                queried_agg_time_dimension_specs=tuple(queried_agg_time_dimension_specs),
                 metric_source_node=unaggregated_measure_node,
             )
 
+        curent_node_time_dimension_specs = self._node_data_set_resolver.get_output_data_set(
```

Comment:
TODO: separate this into its own commit if keeping it. Maybe consider using some alt logic though that doesn't require traversing the whole DAG